### PR TITLE
clk: stm32mp25: Add RIF configuration support

### PR DESCRIPTION
**Pull request by GseoC**
Add RCC RIF configuration support in the driver and the configuration for the stm32mp257f-ev1 board.

### CODE REVIEW

**Comment by GseoC** [comment]
Hello @Gabriel-Fernandz, can you check this P-R please?

**Review by etienne-lms — COMMENTED** [review]
some minor comments

**Comment by Gabriel-Fernandz** [comment]
Hi Gatien,
you can add
Reviewed-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Thank's !

File: core/drivers/clk/clk-stm32mp25.c [hunk 1]
```diff
@@ -3427,6 +3501,210 @@ static struct clk_stm32_priv stm32mp25_clock_data = {
...
+}
+
+static TEE_Result rcc_rif_config(void)
+{
+	TEE_Result res = TEE_ERROR_ACCESS_DENIED;
+	bool is_tdcid = false;
+
+	res = stm32_rifsc_check_tdcid(&is_tdcid);
+	if (res)
+		return res;
```
[the remark this answers is not in the document]
Comment (GseoC) [reply core/drivers/clk/clk-stm32mp25.c:3695]: We could considering the stage at which it's executed.

**Comment by GseoC** [comment]
Comments addressed and tag applied

File: core/drivers/clk/clk-stm32mp25.c [hunk 2]
```diff
@@ -55,6 +64,34 @@
...
+/*
+ * SEMCR register bitfields
+ */
+#define RCC_SEMCR_SCID_MASK		GENMASK_32(6, 4)
+#define RCC_SEMCR_SCID_SHIFT		U(4)
+
+/*
+ * RIF miscellaneous
+ */
+
```
Comment (etienne-lms) [inline core/drivers/clk/clk-stm32mp25.c:91]: Remove this empty line.
By the way, prefer One line inline comments at lines 68, 78, 83 and 89.

File: core/drivers/clk/clk-stm32mp25.c [hunk 3]
```diff
@@ -55,6 +64,34 @@
...
+#define RCC_NB_RIF_RES			U(114)
+#define RCC_NB_CONFS			DIV_ROUND_UP(RCC_NB_RIF_RES, 32)
```
Comment (etienne-lms) [inline core/drivers/clk/clk-stm32mp25.c:93]: Prefer `ROUNDUP_DIV()` (see https://github.com/OP-TEE/optee_os/pull/7116#discussion_r1865575758)

File: core/drivers/clk/clk-stm32mp25.c [hunk 4]
```diff
@@ -3427,6 +3496,210 @@ static struct clk_stm32_priv stm32mp25_clock_data = {
...
+				pdata->conf_data.sec_conf[index]);
+	}
+
+	if (!is_tdcid)
+		goto end;
+
+	for (i = 0; i < RCC_NB_RIF_RES; i++) {
+		index = i / 32;
+
+		if (!(BIT(i % 32) & pdata->conf_data.access_mask[index]))
```
Comment (etienne-lms) [inline core/drivers/clk/clk-stm32mp25.c:3587]: nitpicking: `if (!(BIT(i % 32) & pdata->conf_data.access_mask[i / 32]))`
Suggestion:
```
	for (i = 0; i < RCC_NB_RIF_RES; i++)
		if (BIT(i % 32) & pdata->conf_data.access_mask[i / 32])
			io_clrsetbits32(pdata->rcc_base + RCC_CIDCFGR(i),
					RCC_CIDCFGR_CONF_MASK,
					pdata->conf_data.cid_confs[i]);

File: core/drivers/clk/clk-stm32mp25.c [hunk 5]
```diff
@@ -3427,6 +3496,210 @@ static struct clk_stm32_priv stm32mp25_clock_data = {
...
+			continue;
+
+		io_clrsetbits32(pdata->rcc_base + RCC_CIDCFGR(i),
+				RCC_CIDCFGR_CONF_MASK,
+				pdata->conf_data.cid_confs[i]);
+	}
+
+	for (index = 0; index < RCC_NB_CONFS; index++) {
+		io_clrsetbits32(pdata->rcc_base + RCC_RCFGLOCKR(index),
```
Comment (etienne-lms) [inline core/drivers/clk/clk-stm32mp25.c:3596]: Use `io_setbits32()` for consistency.

File: core/drivers/clk/clk-stm32mp25.c [hunk 6]
```diff
@@ -3427,6 +3496,210 @@ static struct clk_stm32_priv stm32mp25_clock_data = {
...
+static TEE_Result apply_rcc_rif_config(bool is_tdcid)
+{
+	TEE_Result res = TEE_ERROR_ACCESS_DENIED;
+	struct stm32_clk_platdata *pdata = &stm32mp25_clock_pdata;
+	unsigned int i = 0;
+	unsigned int index = 0;
+
+	if (is_tdcid) {
+		for (i = 0; i < RCC_NB_RIF_RES; i++) {
+			index = i / 32;
```
Comment (etienne-lms) [inline core/drivers/clk/clk-stm32mp25.c:3551]: Using `index` here is not really needed.
Suggestion:
```c
	if (is_tdcid) {
		/*
		 * When TDCID, OP-TEE should be the one to set the CID filtering
		 * configuration. Clearing previous configuration prevents
		 * undesired events during the only legitimate configuration.
		 */
		for (i = 0; i < RCC_NB_RIF_RES; i++)
			if (BIT(i % 32) & pdata->conf_data.access_mask[i / 32])
				io_clrbits32(pdata->rcc_base + RCC_CIDCFGR(i),
					     RCC_CIDCFGR_CONF_MASK);
	} else {

**Review by etienne-lms — COMMENTED** [review]
Some comments I previously missed.

**Comment by GseoC** [comment]
Comments addressed

**Review by etienne-lms — COMMENTED** [review]
`Reviewed-by: Etienne Carriere <etienne.carriere@foss.st.com>`

**Comment by GseoC** [comment]
Commits squashed and tag applied, thanks

**Pull request merged**